zanata-war/src/test/java/org/zanata/test/rule/DataSetOperation.java

```diff
@@ -0,0 +1,154 @@
+
+    /**
+     * Defaults to <tt>DatabaseOperation.CLEAN_INSERT</tt>
+     *
+     * @param dataSetLocation
+     *            location of DBUnit dataset
+     */
+    public DataSetOperation(String dataSetLocation) {
+        this(dataSetLocation, DatabaseOperation.CLEAN_INSERT);
+    }
+
+    /**
+     * Defaults to <tt>DatabaseOperation.CLEAN_INSERT</tt>
+     *
+     * @param dataSetLocation
+     *            location of DBUnit dataset
+     * @param dtdLocation
+     *            optional (can be null) location of XML file DTD on
+     *            classpath
+     */
+    public DataSetOperation(String dataSetLocation, String dtdLocation) {
+        this(dataSetLocation, dtdLocation, DatabaseOperation.CLEAN_INSERT);
+    }
+
+    /**
+     * @param dataSetLocation
+     *            location of DBUnit dataset
+     * @param operation
+     *            operation to execute
+     */
+    public DataSetOperation(String dataSetLocation,
+            DatabaseOperation operation) {
+        this(dataSetLocation, null, operation);
+    }
+
+    public DataSetOperation(String dataSetLocation, String dtdLocation,
+            DatabaseOperation operation) {
+        if (dataSetLocation == null) {
+            this.operation = operation;
+            return;
+        }
+
+        // Load the base dataset file
+        InputStream input =
+                Thread.currentThread().getContextClassLoader()
+                        .getResourceAsStream(dataSetLocation);
+        try {
+            FlatXmlDataSetBuilder dataSetBuilder =
+                    new FlatXmlDataSetBuilder();
+            dataSetBuilder.setColumnSensing(true);
+
+            InputStream dtdInput = null;
+            if (dtdLocation != null) {
+                dtdInput =
+                        Thread.currentThread().getContextClassLoader()
+                                .getResourceAsStream(dtdLocation);
+            }
+            if (dtdInput == null) {
+                this.dataSet =
+                        new ReplacementDataSet(dataSetBuilder.build(input));
+            } else {
+                dataSetBuilder.setMetaDataSetFromDtd(dtdInput);
+                this.dataSet =
+                        new ReplacementDataSet(dataSetBuilder.build(input));
+            }
+        } catch (Exception ex) {
+            throw new RuntimeException(ex);
+        }
+        this.operation = operation;
+        this.dataSetLocation = dataSetLocation;
+    }
+
+    public IDataSet getDataSet() {
+        return dataSet;
+    }
+
+    public DatabaseOperation getOperation() {
+        return operation;
+    }
+
+    public void prepare(boolean replaceNull) {
+        if (dataSet == null)
+            return;
+
+        if (replaceNull) {
+            dataSet.addReplacementObject("[NULL]", null);
+        }
+        // TODO This method doesn't implement the functionality of its
+        // original form  (see ZanataDbUnitJpaTest) with regard to the binary
+        // dir replacement.
```

Comment:
Okay. In that case, I'd say get rid of the TODO comment, since we don't
plan to do it.

On 23 March 2016 at 14:44, Carlos Munoz <notifications@github.com> wrote:

> In zanata-war/src/test/java/org/zanata/test/rule/DataSetOperation.java
> <https://github.com/zanata/zanata-server/pull/1107#discussion_r57109370>:
>
> > +    }
> > +
> > +    public DatabaseOperation getOperation() {
> > +        return operation;
> > +    }
> > +
> > +    public void prepare(boolean replaceNull) {
> > +        if (dataSet == null)
> > +            return;
> > +
> > +        if (replaceNull) {
> > +            dataSet.addReplacementObject("[NULL]", null);
> > +        }
> > +        // TODO This method doesn't implement the functionality of its
> > +        // original form  (see ZanataDbUnitJpaTest) with regard to the binary
> > +        // dir replacement.
>
> This is about some functionality present in ZanataDBUnitJpaTest which is
> not ported over to this new class, and which replaces the string
> [binaryDir] with the current locale directory. I did not bring it over as
> I don't think it's being used.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/1107/files/04678b5bd3f9626bca1d1dd92c7e25136727f9c7#r57109370>
>



-- 
Sean Flanigan

Principal Software Engineer
Globalisation Tools Engineering
Red Hat